script.js

```diff
@@ -966,64 +946,103 @@ require([
   // Add the unique values to the
   // section selection element.
   function buildSectionDropdown(values) {
-    var option = domConstruct.create("option");
+    // remove sections
+    for (j = sectionSelect.options.length - 1; j >= 0; j--) {
+      sectionSelect.remove(j);
+    }
+
+    const option = domConstruct.create("option");
     option.text = "Zoom to a Section";
     sectionSelect.add(option);
-
-    values.features.forEach(function (value) {
-      var option = domConstruct.create("option");
-      option.text = value.attributes.sec_ch;
+    const sectionArr = values.features
+    const sectionIntArr = sectionArr.map(element => parseInt(element.attributes.sec_ch));
+    const disctinctSectionsArr = [...new Set(sectionIntArr)];
+    const sortedSections = disctinctSectionsArr.sort((a, b) => a-b).map(element => element.toString().padStart(2, '0'));
+
+    sortedSections.forEach(function (value) {
+      const option = domConstruct.create("option");
+      option.text = value
       sectionSelect.add(option);
     });
   }
 
-  // when township changes, reset the other dropdowns.
-  on(townshipSelect, "change", function (evt) {
-    var type = evt.target.value;
-    var i;
-    for (i = rangeSelect.options.length - 1; i >= 0; i--) {
-      rangeSelect.remove(i);
+  const handleResults = results => {
+    if (results.features.length > 0) {      
+      return results;
+    } else { 
+        $("#trs").prepend(
+          `<div id="TRSAlert" class="alert alert-danger" role="alert">
+            <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+            Invalid Township-Range combination. Please enter a correct Township-Range combination.
+          </div>`
+        )
+        window.setTimeout(function() {
+          $("#TRSAlert").fadeTo(500, 0).slideUp(500, function(){
+              $(this).remove(); 
+          });
+        }, 4000);
+        // remove sections
+        for (j = sectionSelect.options.length - 1; j >= 0; j--) {
+          sectionSelect.remove(j);
+        }
+        throw new Error('This is an invalid Township-Range combination');;
     }
+  }
 
-    var rangeQuery = new Query({
-      where: "twn_ch = '" + type.substr(0, 2) + "' AND tdir = '" + type.substr(2) + "'",
-      outFields: ["rng_ch", "rdir"],
-      returnDistinctValues: true,
-      orderByFields: ["rng_ch", "rdir"]
-    });
-    return townshipRangeSectionLayer.queryFeatures(rangeQuery).then(buildRangeDropdown);
-  })
-
-  // when range changes, reset the section dropdown.
-  on(rangeSelect, "change", function (evt) {
-    var type = evt.target.value;
-    var j;
-    for (j = sectionSelect.options.length - 1; j >= 0; j--) {
-      sectionSelect.remove(j);
+  function queryTR (type, whichDropdown) {
```

Comment:
I kind of want to separate the .then() conditional but aIso think this is very clear in what it's doing instead of trying to smush everything together more.